spec/Supervisor.spec.ts

```diff
@@ -1,7 +1,16 @@
-import {spawn} from "child_process";
+import {ChildProcess, spawn} from "child_process";
 import * as path from "path";
 import {range} from "lodash";
 
+const childExit = (child: ChildProcess) => {
+  return new Promise<void>((resolve) => {
+    child.on("exit", (code: number) => {
+      resolve();
+      expect(code).toEqual(0);
```

Comment:
Should this happen before we resolve the promise?

Reply:
Now that I think of it, I'm not sure how the control flow works when you call `resolve`, does it just mark this promise as successful or does it pass flow back somewhere else?

Reply:
Resolve does not affect control flow :+1:

```javascript
async function inner() {
  return await new Promise((resolve) => {
    console.log('inner-1');
    resolve();
    console.log('inner-2');
  });
}

async function outer() {
  console.log('outer-1');
  await inner();
  console.log('outer-2');
}

await outer()
```